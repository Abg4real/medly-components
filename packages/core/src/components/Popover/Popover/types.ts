import { HTMLProps } from '@medly-components/utils';

export interface Props extends HTMLProps<HTMLDivElement> {
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
}
