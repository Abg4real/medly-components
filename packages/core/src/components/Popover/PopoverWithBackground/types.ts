import { HTMLProps } from '@medly-components/utils';
import { Placement } from '../types';

export interface Props extends HTMLProps<HTMLDivElement> {
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
    /** Function to be called on  outer click of the wrapper */
    placement?: Placement;
}
