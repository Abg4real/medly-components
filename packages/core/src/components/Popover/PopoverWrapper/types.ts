import { HTMLProps, Omit } from '@medly-components/utils';

export type Placement =
    | 'top-start'
    | 'top'
    | 'top-end'
    | 'right-start'
    | 'right'
    | 'right-end'
    | 'bottom-end'
    | 'bottom'
    | 'bottom-start'
    | 'left-end'
    | 'left'
    | 'left-start';

export type InteractionType = 'click' | 'hover';

export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'onChange'> {
    /** Pass true to show popover child */
    showPopover?: boolean;
    /** Placement of the popover */
    placement?: Placement;
    /** Function to be called on  outer click of the wrapper */
    onOuterClick?: () => void;
}
