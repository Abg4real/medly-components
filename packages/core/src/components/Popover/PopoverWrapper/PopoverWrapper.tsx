import { useOuterClickNotifier, WithStyle } from '@medly-components/utils';
import React, { SFC, useCallback, useEffect, useRef } from 'react';
import { PopoverWrapperStyled } from './PopoverWrapper.styled';
import { Props } from './types';

export const PopoverWrapper: SFC<Props> & WithStyle = React.memo(props => {
    const wrapperRef = useRef(null),
        { onOuterClick, showPopover, ...restProps } = props,
        id = props.id || 'medly-popover-wrapper';

    const changePopoverState = useCallback(
        (state: boolean) => {
            if (props.interactionType === 'click') {
                const popoverIndex = wrapperRef.current.children.length - 1,
                    popover = wrapperRef.current.children[popoverIndex] as HTMLElement;
                popover.style.display = state ? 'block' : 'none';
            }
        },
        [props.interactionType, wrapperRef.current]
    );

    useEffect(() => {
        changePopoverState(showPopover);
    }, [showPopover]);

    useOuterClickNotifier(() => {
        changePopoverState(false);
        onOuterClick && onOuterClick();
    }, wrapperRef);

    return <PopoverWrapperStyled {...restProps} ref={wrapperRef} id={id} />;
});

PopoverWrapper.displayName = 'PopoverWrapper';
PopoverWrapper.defaultProps = {
    interactionType: 'hover',
    placement: 'bottom'
};
PopoverWrapper.Style = PopoverWrapperStyled;
