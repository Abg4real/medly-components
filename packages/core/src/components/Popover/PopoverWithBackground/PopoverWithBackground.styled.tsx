import { css, fullHeight, fullWidth, styled } from '@medly-components/utils';
import { Placement } from '../PopoverWrapper/types';
import { Props } from './types';

const getContainerMargin = (placement: Placement) => {
    switch (placement) {
        case 'top':
        case 'top-start':
        case 'top-end':
            return css`
                bottom: calc(100% + 15px);
            `;
        case 'right':
        case 'right-start':
        case 'right-end':
            return css`
                left: calc(100% + 15px);
            `;
        case 'bottom':
        case 'bottom-start':
        case 'bottom-end':
            return css`
                top: calc(100% + 15px);
            `;
        case 'left':
        case 'left-start':
        case 'left-end':
            return css`
                right: calc(100% + 15px);
            `;
    }
};

const getRectangleShadow = (placement: Placement) => {
    switch (placement) {
        case 'top':
        case 'top-start':
        case 'top-end':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
            `;
        case 'right':
        case 'right-start':
        case 'right-end':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
            `;
        case 'bottom':
        case 'bottom-start':
        case 'bottom-end':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
            `;
        case 'left':
        case 'left-start':
        case 'left-end':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
            `;
    }
};

const getTriangleStyle = (placement: Placement) => {
    switch (placement) {
        case 'top':
        case 'top-start':
        case 'top-end':
            return css`
                transform: translateX(-50%);
                top: 100%;
                left: 50%;
            `;
        case 'right':
        case 'right-start':
        case 'right-end':
            return css`
                transform: translateY(-50%) rotate(90deg);
                top: 50%;
                right: calc(100% - 3px);
            `;
        case 'bottom':
        case 'bottom-start':
        case 'bottom-end':
            return css`
                transform: translateX(-50%) rotate(180deg);
                left: 50%;
                bottom: 100%;
            `;
        case 'left':
        case 'left-start':
        case 'left-end':
            return css`
                transform: translateY(-50%) rotate(270deg);
                top: 50%;
                left: calc(100% - 3px);
            `;
    }
};

export const PopoverStyled = styled('div')<Props>`
    position: absolute;
    z-index: 1000;
    transition: all 2ms ease-in-out;
    align-self: center;
    ${props => props.fullWidth && fullWidth()};
    ${props => props.fullHeight && fullHeight()};
    ${({ placement }) => getContainerMargin(placement)}
`;

export const PopoverBackground = styled('div')<Props>`
    z-index: -1;
    position: absolute;
    width: 100%;
    height: 100%;
`;

export const Rectangle = styled('div')<Props>`
    background-color: white;
    border-radius: 8px;
    ${({ placement }) => getRectangleShadow(placement)}
    width: 100%;
    height: 100%;
    position: relative;
    &::after {
        width: 0;
        height: 0;
        z-index: 10;
        position: absolute;
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        border-top: 10px solid #fff;
        ${({ placement }) => getTriangleStyle(placement)}
        content: '';
    }
`;

export const Triangle = styled('div')<Props>``;

export const PopoverContent = styled('div')`
    padding: 15px 20px 15px 20px;
`;
