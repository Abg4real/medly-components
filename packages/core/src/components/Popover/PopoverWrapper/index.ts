import { PopoverWrapper } from './PopoverWrapper';

export default PopoverWrapper;
