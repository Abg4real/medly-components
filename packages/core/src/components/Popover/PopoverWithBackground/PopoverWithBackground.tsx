import { WithStyle } from '@medly-components/utils';
import React, { FC } from 'react';
import { PopoverBackground, PopoverContent, PopoverStyled, Rectangle } from './PopoverWithBackground.styled';
import { Props } from './types';

export const PopoverWithBackground: FC<Props> & WithStyle = React.memo(props => (
    <PopoverStyled {...props} id={props.id || 'medly-popover'}>
        <PopoverBackground placement={props.placement}>
            <Rectangle placement={props.placement} />
        </PopoverBackground>
        <PopoverContent>{props.children}</PopoverContent>
    </PopoverStyled>
));
PopoverWithBackground.displayName = 'PopoverWithBackground';
PopoverWithBackground.Style = PopoverStyled;
PopoverWithBackground.defaultProps = {
    fullWidth: false,
    fullHeight: false,
    placement: 'top'
};
