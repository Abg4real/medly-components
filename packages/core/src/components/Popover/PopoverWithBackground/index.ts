import { PopoverWithBackground } from './PopoverWithBackground';
export default PopoverWithBackground;
