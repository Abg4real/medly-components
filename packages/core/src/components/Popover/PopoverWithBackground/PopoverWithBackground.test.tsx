import { cleanup, render } from '@test-utils';
import React from 'react';
import { Placement } from '../PopoverWrapper/types';
import { PopoverWithBackground } from './PopoverWithBackground';
import { Props } from './types';

const renderer = ({ placement = 'left' }: Props) =>
    render(
        <PopoverWithBackground placement={placement} fullWidth={true} fullHeight={true}>
            Dummy popover
        </PopoverWithBackground>
    );

describe('Popover with background component', () => {
    afterEach(cleanup);

    test.each([
        'top-start',
        'top',
        'top-end',
        'right-start',
        'right',
        'right-end',
        'bottom-end',
        'bottom',
        'bottom-start',
        'left-end',
        'left',
        'left-start'
    ])('%s, should render properly', (placement: Placement) => {
        const { container } = renderer({ placement });
        expect(container).toMatchSnapshot();
    });
});
