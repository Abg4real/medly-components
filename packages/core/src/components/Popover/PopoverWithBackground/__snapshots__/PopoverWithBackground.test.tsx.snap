// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover with background component bottom, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  top: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 -0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%) rotate(180deg);
  -ms-transform: translateX(-50%) rotate(180deg);
  transform: translateX(-50%) rotate(180deg);
  left: 50%;
  bottom: 100%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component bottom-end, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  top: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 -0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%) rotate(180deg);
  -ms-transform: translateX(-50%) rotate(180deg);
  transform: translateX(-50%) rotate(180deg);
  left: 50%;
  bottom: 100%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component bottom-start, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  top: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 -0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%) rotate(180deg);
  -ms-transform: translateX(-50%) rotate(180deg);
  transform: translateX(-50%) rotate(180deg);
  left: 50%;
  bottom: 100%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component left, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  right: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(270deg);
  -ms-transform: translateY(-50%) rotate(270deg);
  transform: translateY(-50%) rotate(270deg);
  top: 50%;
  left: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component left-end, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  right: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(270deg);
  -ms-transform: translateY(-50%) rotate(270deg);
  transform: translateY(-50%) rotate(270deg);
  top: 50%;
  left: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component left-start, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  right: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(270deg);
  -ms-transform: translateY(-50%) rotate(270deg);
  transform: translateY(-50%) rotate(270deg);
  top: 50%;
  left: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component right, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  left: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: -0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(90deg);
  -ms-transform: translateY(-50%) rotate(90deg);
  transform: translateY(-50%) rotate(90deg);
  top: 50%;
  right: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component right-end, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  left: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: -0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(90deg);
  -ms-transform: translateY(-50%) rotate(90deg);
  transform: translateY(-50%) rotate(90deg);
  top: 50%;
  right: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component right-start, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  left: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: -0.4rem 0 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateY(-50%) rotate(90deg);
  -ms-transform: translateY(-50%) rotate(90deg);
  transform: translateY(-50%) rotate(90deg);
  top: 50%;
  right: calc(100% - 3px);
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component top, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  bottom: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  top: 100%;
  left: 50%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component top-end, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  bottom: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  top: 100%;
  left: 50%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover with background component top-start, should render properly 1`] = `
.c0 {
  position: absolute;
  z-index: 1000;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: calc(100% - 0px);
  height: calc(100% - 0px);
  bottom: calc(100% + 15px);
}

.c1 {
  z-index: -1;
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 0.4rem 1.6rem rgba(176,188,200,0.5);
  width: 100%;
  height: 100%;
  position: relative;
}

.c2::after {
  width: 0;
  height: 0;
  z-index: 10;
  position: absolute;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-top: 10px solid #fff;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  top: 100%;
  left: 50%;
  content: '';
}

.c3 {
  padding: 15px 20px 15px 20px;
}

<div>
  <div
    class="c0"
    id="medly-popover"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      Dummy popover
    </div>
  </div>
</div>
`;
